sdk/lib/crt/setjmp/amd64/setjmp.s

```diff
@@ -135,35 +143,34 @@ FUNC longjmp
 
     .endprolog
 
-    // FIXME: handle frame
-
-    mov rbx, [rcx + JUMP_BUFFER_Rbx]
-    mov rbp, [rcx + JUMP_BUFFER_Rbp]
-    mov rsi, [rcx + JUMP_BUFFER_Rsi]
-    mov rdi, [rcx + JUMP_BUFFER_Rdi]
-    mov r12, [rcx + JUMP_BUFFER_R12]
-    mov r13, [rcx + JUMP_BUFFER_R13]
-    mov r14, [rcx + JUMP_BUFFER_R14]
-    mov r15, [rcx + JUMP_BUFFER_R15]
-    mov rsp, [rcx + JUMP_BUFFER_Rsp]
-    mov r8, [rcx + JUMP_BUFFER_Rip]
-    movdqa xmm6, [rcx + JUMP_BUFFER_Xmm6]
-    movdqa xmm7, [rcx + JUMP_BUFFER_Xmm7]
-    movdqa xmm8, [rcx + JUMP_BUFFER_Xmm8]
-    movdqa xmm9, [rcx + JUMP_BUFFER_Xmm9]
-    movdqa xmm10, [rcx + JUMP_BUFFER_Xmm10]
-    movdqa xmm11, [rcx + JUMP_BUFFER_Xmm11]
-    movdqa xmm12, [rcx + JUMP_BUFFER_Xmm12]
-    movdqa xmm13, [rcx + JUMP_BUFFER_Xmm13]
-    movdqa xmm14, [rcx + JUMP_BUFFER_Xmm14]
-    movdqa xmm15, [rcx + JUMP_BUFFER_Xmm15]
-
-    /* return param2 or 1 if it was 0 */
-    mov rax, rdx
-    test rax, rax
-    jnz l2
-    inc rax
-l2: jmp r8
+    mov rbx, [rcx + JUMP_BUFFER_Rbx]            /* Restore rbx */
+    mov rsp, [rcx + JUMP_BUFFER_Rsp]            /* Restore rsp */
+    mov rbp, [rcx + JUMP_BUFFER_Rbp]            /* Restore rbp */
+    mov rsi, [rcx + JUMP_BUFFER_Rsi]            /* Restore rsi */
+    mov rdi, [rcx + JUMP_BUFFER_Rdi]            /* Restore rdi */
+    mov r12, [rcx + JUMP_BUFFER_R12]            /* Restore r12 */
+    mov r13, [rcx + JUMP_BUFFER_R13]            /* Restore r13 */
+    mov r14, [rcx + JUMP_BUFFER_R14]            /* Restore r14 */
+    mov r15, [rcx + JUMP_BUFFER_R15]            /* Restore r15 */
+    mov rax, [rcx + JUMP_BUFFER_Frame]          /* Get frame pointer */
+    mov [rsp + 8], rax                          /* Restore frame pointer */
+    movdqu xmm6, [rcx + JUMP_BUFFER_Xmm6]       /* Restore xmm6 */
+    movdqu xmm7, [rcx + JUMP_BUFFER_Xmm7]       /* Restore xmm7 */
+    movdqu xmm8, [rcx + JUMP_BUFFER_Xmm8]       /* Restore xmm8 */
+    movdqu xmm9, [rcx + JUMP_BUFFER_Xmm9]       /* Restore xmm9 */
+    movdqu xmm10, [rcx + JUMP_BUFFER_Xmm10]     /* Restore xmm10 */
+    movdqu xmm11, [rcx + JUMP_BUFFER_Xmm11]     /* Restore xmm11 */
+    movdqu xmm12, [rcx + JUMP_BUFFER_Xmm12]     /* Restore xmm12 */
+    movdqu xmm13, [rcx + JUMP_BUFFER_Xmm13]     /* Restore xmm13 */
+    movdqu xmm14, [rcx + JUMP_BUFFER_Xmm14]     /* Restore xmm14 */
+    movdqu xmm15, [rcx + JUMP_BUFFER_Xmm15]     /* Restore xmm15 */
+    mov rax, rdx                                /* Move val into rax (return value) */
+    test rax, rax                               /* Check if val is 0 */
+    jz LABEL3                                   /* If val is 0, jump to LABEL3 */
+    jmp qword ptr [rcx + JUMP_BUFFER_Rip]       /* Jump to the stored return address (rip) */
+LABEL3:
+    mov rax, 1                                  /* If val was 0, return 1 */
```

Comment:
Note:
* `i386` code misses this case. To be fixed.
* `arm` code always returns `1`. To be improved.

Reply:
Seems fixed.